src/Symfony/Component/Messenger/Adapter/AmqpExt/Connection.php

```diff
@@ -101,6 +101,86 @@ public function publish(string $body, array $headers = array()): void
         $this->exchange()->publish($body, null, AMQP_NOPARAM, array('headers' => $headers));
     }
 
+    /**
+     * @throws \AMQPException
+     */
+    public function publishForRetry(\AMQPEnvelope $message, array $retryConfiguration): bool
+    {
+        $attemptNumber = $message->getHeader('symfony-messenger-attempts') ?: 0;
+
+        if ($this->shouldSetup()) {
+            $this->setupRetry($retryConfiguration, $attemptNumber);
+        }
+
+        $maximumAttempts = $retryConfiguration['attempts'] ?? 3;
+        $routingKey = str_replace('%attempt%', $attemptNumber, $retryConfiguration['routing_key_pattern'] ?? 'attempt_%attempt%');
+
+        if ($attemptNumber > $maximumAttempts) {
+            if (!isset($retryConfiguration['dead_queue'])) {
+                return false;
+            }
+
+            $routingKey = 'dead';
```

Comment:
Should be a configurable `dead_routing_key`

Reply:
Well, that’s the term used by AMQP unfortunately. Maybe `too_many_attempts` if really you want to change? 🤔